### PR TITLE
Add Swagger Specs for Azure Stream Analytics 2016-03-01 API version

### PR DESCRIPTION
**Pull request by atpham256**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by veronicagg** [comment]
Adding @ravbhatnagar as a reviewer since this is a new spec, and he's been involved in the conversation. Gaurav, please take a look from your side and let me know.

**Comment by veronicagg** [comment]
@atpham256 Thanks for submitting a clean PR! Just a couple of minor questions, and pulling in Gaurav from the ARM team to approve the PR. Once that's done, I can approve and merge. thanks!

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: arm-streamanalytics/2016-03-01/swagger/transformations.json [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Transformation": {
```
Comment (atpham256) [inline arm-streamanalytics/2016-03-01/swagger/transformations.json]: @veronicagg Correct. Transformation has no location property associated with it. This is also true for Inputs, Outputs, and Functions.

File: arm-streamanalytics/2016-03-01/swagger/functions.json [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+            "schema": {
+              "$ref": "#/definitions/Function"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Function": {
```
Comment (atpham256) [inline arm-streamanalytics/2016-03-01/swagger/functions.json]: @veronicagg If you are referring to deep PUT, that would be in streamingjobs.json. Streaming jobs are our top-level resource with inputs, outputs, transformation, and functions being nested-resources under it.

**Comment by atpham256** [comment]
Ping @veronicagg @ravbhatnagar :)

**Comment by ravbhatnagar** [comment]
@atpham256  - As discussed over email, the only major issue is around doing a PUT on parent resource which also creates child resources. Please note that once ARM supports Deep PUT natively, your team will have to change this implementation to adopt to the new model. Please confirm that you are fine with this and ARM team can sign off on this.

**Comment by atpham256** [comment]
@ravbhatnagar Of course. Once ARM supports Deep PUT natively, we will adopt that model. I imagine it will be a requirement in the ARM Rest API contract so it's not like we would have a choice. :)

**Comment by ravbhatnagar** [comment]
@atpham256 - look good from ARM side. @veronicagg - ARM signing off.

**Comment by veronicagg** [comment]
thanks @ravbhatnagar .

**Review by veronicagg — APPROVED** [review]

**Pull request merged**